src/ServiceAccount/Search/ClientIDFilter.tsx

```diff
@@ -0,0 +1,116 @@
+import { FilterProps } from './FilterProps';
+import React, { useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { TooltipContent } from './TooltipContent'
+import {
+  Button,
+  ButtonVariant,
+  InputGroup,
+  TextInput,
+  ToolbarFilter,
+  Tooltip,
+  ValidatedOptions,
+} from '@patternfly/react-core';
+import SearchIcon from '@patternfly/react-icons/dist/js/icons/search-icon';
+
+export type ClientIDFilterProps = FilterProps;
+
+export const ClientIDFilter: React.FunctionComponent<ClientIDFilterProps> = ({
+  getSelectionForFilter,
+  onDeleteChipGroup,
+  onDeleteChip,
+  filterSelected,
+  isMaxFilter,
+  updateFilter,
+}) => {
+  const { t } = useTranslation();
+  const inputRef = useRef<HTMLInputElement>(null);
+  const tooltipContent = TooltipContent(isMaxFilter, 'clientid');
+
+  const [value, setValue] = useState<string | undefined>();
```

Comment:
The `value` should be a prop, not a state of this component. https://overreacted.io/writing-resilient-components/#principle-1-dont-stop-the-data-flow